Install profile

dannylamb opened the pull request:
Pulls in all the work to use Born Digital's new install profile.

dannylamb committed:
Nerfing make demo so I can get through to actually using the install profile

dannylamb committed:
Finding profile and blowing up on not finding the theme

dannylamb committed:
Finally installing

dannylamb committed:
Making it so you can build from images on dockerhub

dannylamb committed:
Updating makefile

dannylamb committed:
updating buildkit hash

dannylamb committed:
Make local works

Commit:
Confirm on clean

dannylamb committed:
Adding REPOSITORY to make local

dannylamb committed:
Bumping buildkit hash

Commit:
Adds the config/sync directory if it doesn't already exist and chowns it

Adds the correct permissions to the sync directory as well.

dannylamb committed:
Getting ready to pull in upstream

seth-shaw-unlv commented:
I just tried this out on a clean checkout of isle-dc and it didn't complete successfully:
```
$ git checkout -b install-profile origin/install-profile
Branch install-profile set up to track remote branch install-profile from origin.
Switched to a new branch 'install-profile'
$ cp sample.env .env
$ make local
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  5596  100  5596    0     0   8757      0 --:--:-- --:--:-- --:--:--  8743
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  1704  100  1704    0     0   5958      0 --:--:-- --:--:-- --:--:--  5958
docker-compose pull
Pulling activemq   ... done
Pulling alpaca     ... done
Pulling blazegraph ... done
Pulling cantaloupe ... done
Pulling fcrepo     ... done
Pulling fits       ... done
Pulling crayfits   ... done
Pulling homarus    ... done
Pulling houdini    ... done
Pulling hypercube  ... done
Pulling mariadb    ... done
Pulling matomo     ... done
Pulling milliner   ... done
Pulling recast     ... done
Pulling solr       ... done
Pulling traefik    ... done
Pulling drupal     ... done
Unable to find image 'islandora/nginx:c2d65aee856887b49b2cbf889060e549a83e4946' locally
c2d65aee856887b49b2cbf889060e549a83e4946: Pulling from islandora/nginx
Digest: sha256:7df17a78c0554243bee046c81bdb0ac1b5bcfa16fe2a2930fbf94744c9aba6ee
Status: Downloaded newer image for islandora/nginx:c2d65aee856887b49b2cbf889060e549a83e4946
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] 00-container-environment-00-init.sh: executing... 
[cont-init.d] 00-container-environment-00-init.sh: exited 0.
[cont-init.d] 00-container-environment-01-override-nginx.sh: executing... 
[cont-init.d] 00-container-environment-01-override-nginx.sh: exited 0.
[cont-init.d] 00-container-environment-01-override-php.sh: executing... 
[cont-init.d] 00-container-environment-01-override-php.sh: exited 0.
[cont-init.d] 00-container-environment-02-database-defaults.sh: executing... 
[cont-init.d] 00-container-environment-02-database-defaults.sh: exited 0.
[cont-init.d] 01-confd-render-templates.sh: executing... 
[cont-init.d] 01-confd-render-templates.sh: exited 0.
[cont-init.d] 02-fpm-install.sh: executing... 
[cont-init.d] 02-fpm-install.sh: exited 0.
[cont-init.d] 02-ngnix-install.sh: executing... 
[cont-init.d] 02-ngnix-install.sh: exited 0.
[cont-init.d] done.
[services.d] starting services
[services.d] done.
Cloning into '/tmp/codebase'...
[27-Jul-2021 18:01:23] NOTICE: fpm is running, pid 727
[27-Jul-2021 18:01:23] NOTICE: ready to handle connections

                                                                                                                                                                                                                                                                                                                                             
  [InvalidArgumentException]                                                                                                                                                                                                                                                                                                                 
  Package islandora/islandora_profile exists in composer repo (https://repo.packagist.org) and package repo (defining 1 package) which has a higher repository priority. The packages with higher priority do not match your constraint and are therefore not installable. See https://getcomposer.org/repoprio for details and assistance.  
                                                                                                                                                                                                                                                                                                                                             

require [--dev] [--dry-run] [--prefer-source] [--prefer-dist] [--fixed] [--no-suggest] [--no-progress] [--no-update] [--no-install] [--no-scripts] [--update-no-dev] [-w|--update-with-dependencies] [-W|--update-with-all-dependencies] [--with-dependencies] [--with-all-dependencies] [--ignore-platform-req IGNORE-PLATFORM-REQ] [--ignore-platform-reqs] [--prefer-stable] [--prefer-lowest] [--sort-packages] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--apcu-autoloader-prefix APCU-AUTOLOADER-PREFIX] [--] [<packages>]...

[cmd] with-contenv exited 1
[27-Jul-2021 18:01:32] NOTICE: Terminating ...
[27-Jul-2021 18:01:32] NOTICE: exiting, bye-bye!
[cont-finish.d] executing container finish scripts...
[cont-finish.d] done.
[s6-finish] waiting for services.
[services.d] service fpm finish: executing...
[services.d] service fpm exiting with exit code: 0
s6-svwait: fatal: supervisor died
[services.d] service nginx finish: executing...
[services.d] service nginx exiting with exit code: 0
[s6-finish] sending all processes the TERM signal.
[s6-finish] sending all processes the KILL signal and exiting.
make: *** [local] Error 1
```

noahwsmith committed:
Temp edit - islandora_profile in packagist

This can be removed once this PR is resolved https://github.com/Islandora-Devops/islandora-project/pull/2

noahwsmith commented:
@seth-shaw-unlv try again- I just pushed a small patch for this issue. @dannylamb put install_profile in packagist, which means we can remove the package declaration https://github.com/Islandora-Devops/islandora-project/pull/2

noahwsmith committed:
patching Makefile for d9

noahwsmith committed:
cleanup on install

noahwsmith commented:
Status update - @dannylamb is going to get this back to the point where Composer installs cleanly, and then I'll jump back in and see if all the configs are landing correctly.

dannylamb commented:
This works with `make demo` but not with `make local`.  I'm currently investigating what the difference is between the two codebases and trying to bring them into alignment.

dannylamb committed:
Bringing make local and make demo codebases into alignment

dannylamb commented:
This works now with `make local`. Haven't tested anything super thoroughly, but the site comes up now.

The issue was with trying to use https://github.com/islandora-devops/islandora-project.  There's some additional things I added to the composer.json in https://github.com/Islandora-Devops/islandora-sandbox that would need to be added.  Mostly just stuff around the config sync / config split directories.

noahwsmith commented:
I can confirm that this works - thanks @dannylamb !

@seth-shaw-unlv this is ready for re-testing. Just `git pull`, `sudo rm -rf codebase` and `make local` and you should be back in action.

noahwsmith commented:
Actually, if we can get confirmation that this is in a good working state, I'd love to move forward with merging PR...

seth-shaw-unlv commented:
I started it up yesterday but it appeared to hang and I didn't get to take a closer look before I left for the day. Unfortunately I'm now stuck at home until we can confirm that the fever one of my kids developed isn't COVID. I can test as soon as I can get back to the office, but if someone else beats me to it, that would be great.

yamensaleh commented:
Hello Seth

Hope ur kid is feeling better.

Have you tested this?

I did but it gave me error on installing modules?

Thanks

On Wed, 4 Aug 2021 at 10:21 PM Seth Shaw ***@***.***> wrote:

> I started it up yesterday but it appeared to hang and I didn't get to take
> a closer look before I left for the day. Unfortunately I'm now stuck at
> home until we can confirm that the fever one of my kids developed isn't
> COVID. I can test as soon as I can get back to the office, but if someone
> else beats me to it, that would be great.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Islandora-Devops/isle-dc/pull/160#issuecomment-892911197>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AD2TQTNLZ254FQGLTMG7FF3T3GHK3ANCNFSM44C3PLMQ>
> .
>
-- 
*Yamen Saleh*

yamensaleh commented:
Actually the error is on run Islandora Migrations, (output below)


In bootstrap.inc line 230:

  The configuration directory type 'sync' does not exist


site:install [--db-url DB-URL] [--db-prefix DB-PREFIX] [--db-su DB-SU]
[--db-su-pw DB-SU-PW] [--account-name [ACCOUNT-NAME]] [--account-mail
[ACCOUNT-MAIL]] [--site-mail [SITE-MAIL]] [--account-pass ACCOUNT-PASS]
[--locale [LOCALE]] [--site-name [SITE-NAME]] [--site-pass SITE-PASS]
[--sites-subdir SITES-SUBDIR] [--config-dir CONFIG-DIR] [--existing-config]
[-h|--help] [-q|--quiet] [-v|vv|vvv|--verbose] [-V|--version] [--ansi]
[--no-ansi] [-n|--no-interaction] [-d|--debug] [-y|--yes] [--no]
[--remote-host REMOTE-HOST] [--remote-user REMOTE-USER] [-r|--root ROOT]
[-l|--uri URI] [--simulate] [--pipe] [-D|--define DEFINE] [--xh-link
XH-LINK] [--druplicon] [--notify [NOTIFY]] [--] <command> [<profile>]...

[sudo] password for yamen:



  Command pm:enable was not found. Drush was unable to query the database.
As
   a result, many commands are unavailable. Re-run your command with
--debug
  to see relevant log messages.







  Command config:set was not found. Drush was unable to query the database.
A
  s a result, many commands are unavailable. Re-run your command with
--debug
   to see relevant log messages.







  Command config:set was not found. Drush was unable to query the database.
A
  s a result, many commands are unavailable. Re-run your command with
--debug
   to see relevant log messages.







  Command role:list was not found. Drush was unable to query the database.
As
   a result, many commands are unavailable. Re-run your command with
--debug
  to see relevant log messages.







  Command role:create was not found. Drush was unable to query the
database.
  As a result, many commands are unavailable. Re-run your command with
--debu
  g to see relevant log messages.







  Command user:role:add was not found. Drush was unable to query the
database
  . As a result, many commands are unavailable. Re-run your command with
--de
  bug to see relevant log messages.




make[1]: [Makefile:137: update-config-from-environment] Error 1 (ignored)



  Command pm:enable was not found. Drush was unable to query the database.
As
   a result, many commands are unavailable. Re-run your command with
--debug
  to see relevant log messages.







  Command config:import was not found. Drush was unable to query the
database
  . As a result, many commands are unavailable. Re-run your command with
--de
  bug to see relevant log messages.




make[1]: [Makefile:138: update-config-from-environment] Error 1 (ignored)



  Command pm-list was not found. Drush was unable to query the database. As
a
   result, many commands are unavailable. Re-run your command with --debug
to
   see relevant log messages.







  Command pm:enable was not found. Drush was unable to query the database.
As
   a result, many commands are unavailable. Re-run your command with
--debug
  to see relevant log messages.







  Command pm:uninstall was not found. Drush was unable to query the
database.
   As a result, many commands are unavailable. Re-run your command with
--deb
  ug to see relevant log messages.




make[1]: [Makefile:140: update-config-from-environment] Error 1 (ignored)



  Command pm-list was not found. Drush was unable to query the database. As
a
   result, many commands are unavailable. Re-run your command with --debug
to
   see relevant log messages.







  Command pm-list was not found. Drush was unable to query the database. As
a
   result, many commands are unavailable. Re-run your command with --debug
to
   see relevant log messages.







  Command pm-list was not found. Drush was unable to query the database. As
a
   result, many commands are unavailable. Re-run your command with --debug
to
   see relevant log messages.







  Command pm-list was not found. Drush was unable to query the database. As
a
   result, many commands are unavailable. Re-run your command with --debug
to
   see relevant log messages.




Waiting for 80 on blazegraph to open.
  % Total    % Received % Xferd  Average Speed   Time    Time     Time
 Current
                                 Dload  Upload   Total   Spent    Left
 Speed
100   736  100    18  100   718     34   1359 --:--:-- --:--:-- --:--:--
 1391
  % Total    % Received % Xferd  Average Speed   Time    Time     Time
 Current
                                 Dload  Upload   Total   Spent    Left
 Speed
100   283  100    61  100   222     36    132  0:00:01  0:00:01 --:--:--
169



  Command migrate:import was not found. Drush was unable to query the
databas
  e. As a result, many commands are unavailable. Re-run your command with
--d
  ebug to see relevant log messages.




make[1]: *** [Makefile:149: run-islandora-migrations] Error 1
make: *** [Makefile:354: local] Error 2

On Sat, Aug 7, 2021 at 2:54 PM Yamen Saleh ***@***.***> wrote:

> Hello Seth
>
> Hope ur kid is feeling better.
>
> Have you tested this?
>
> I did but it gave me error on installing modules?
>
> Thanks
>
> On Wed, 4 Aug 2021 at 10:21 PM Seth Shaw ***@***.***> wrote:
>
>> I started it up yesterday but it appeared to hang and I didn't get to
>> take a closer look before I left for the day. Unfortunately I'm now stuck
>> at home until we can confirm that the fever one of my kids developed isn't
>> COVID. I can test as soon as I can get back to the office, but if someone
>> else beats me to it, that would be great.
>>
>> —
>> You are receiving this because you are subscribed to this thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Islandora-Devops/isle-dc/pull/160#issuecomment-892911197>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AD2TQTNLZ254FQGLTMG7FF3T3GHK3ANCNFSM44C3PLMQ>
>> .
>>
> --
> *Yamen Saleh*
>


-- 
*Yamen Saleh*

seth-shaw-unlv commented:
He is feeling better, thanks! Fortunately it wasn't COVID.

That said, I haven't been back to my office yet where I was testing this. I can give it a go on Monday.

seth-shaw-unlv commented:
I came back to the office this morning and realized that it hadn't completed running before I left the office and was still waiting for a password. So I had it finish running. This run did complete finally and I didn't have @yamensaleh's error. (See the attached terminal output: [isle-born-digital.log](https://github.com/Islandora-Devops/isle-dc/files/6956503/isle-born-digital.log).)

It appears to be working although there are some oddities about it:

Going to the front page just after it finished gave me a 404:
<img width="1028" alt="Screen Shot 2021-08-09 at 11 17 29 AM" src="https://user-images.githubusercontent.com/29869988/128754836-9fcdcda8-79a7-45d3-ad01-948aa4619d7d.png">

And after I created my first object it became my homepage:
<img width="1302" alt="Screen Shot 2021-08-09 at 11 23 38 AM" src="https://user-images.githubusercontent.com/29869988/128755555-b0285268-e105-4eca-9db8-264643072f0f.png">

Also, I had to use the OpenSeadragon display hint because without it we only see the original file download link, rather than the Jpeg service file:
<img width="1300" alt="Screen Shot 2021-08-09 at 11 26 03 AM" src="https://user-images.githubusercontent.com/29869988/128755569-a0af971b-e2f3-4729-9b7f-2a33d6163665.png">
<img width="926" alt="Screen Shot 2021-08-09 at 11 26 50 AM" src="https://user-images.githubusercontent.com/29869988/128755582-c5668ef5-829f-41fd-be4c-527312df5e7b.png">

yamensaleh commented:
Yes, finally it worked, but i had to use the demo values with the
local environment in the .env file to get it working properly

ENVIRONMENT=local
# If you're just demoing or are starting from scratch, use this.
INSTALL_EXISTING_CONFIG=false
DRUPAL_INSTALL_PROFILE=islandora_install_profile_demo

and i have the same scenario Seth had

On Mon, Aug 9, 2021 at 9:31 PM Seth Shaw ***@***.***> wrote:

> I came back to the office this morning and realized that it hadn't
> completed running before I left the office and was still waiting for a
> password. So I had it finish running. This run did complete finally and I
> didn't have @yamensaleh <https://github.com/yamensaleh>'s error. (See the
> attached terminal output: isle-born-digital.log
> <https://github.com/Islandora-Devops/isle-dc/files/6956503/isle-born-digital.log>
> .)
>
> It appears to be working although there are some oddities about it:
>
> Going to the front page just after it finished gave me a 404:
> [image: Screen Shot 2021-08-09 at 11 17 29 AM]
> <https://user-images.githubusercontent.com/29869988/128754836-9fcdcda8-79a7-45d3-ad01-948aa4619d7d.png>
>
> And after I created my first object it became my homepage:
> [image: Screen Shot 2021-08-09 at 11 23 38 AM]
> <https://user-images.githubusercontent.com/29869988/128755555-b0285268-e105-4eca-9db8-264643072f0f.png>
>
> Also, I had to use the OpenSeadragon display hint because without it we
> only see the original file download link, rather than the Jpeg service file:
> [image: Screen Shot 2021-08-09 at 11 26 03 AM]
> <https://user-images.githubusercontent.com/29869988/128755569-a0af971b-e2f3-4729-9b7f-2a33d6163665.png>
> [image: Screen Shot 2021-08-09 at 11 26 50 AM]
> <https://user-images.githubusercontent.com/29869988/128755582-c5668ef5-829f-41fd-be4c-527312df5e7b.png>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Islandora-Devops/isle-dc/pull/160#issuecomment-895444262>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AD2TQTPTNUQ2SJKBQUKUMFTT4ANGLANCNFSM44C3PLMQ>
> .
>


-- 
*Yamen Saleh*

DonRichards commented:
Brought in upstream changes to resolve conflicts.

kstapelfeldt commented:
See today's discussion for more details: https://github.com/Islandora/documentation/wiki/October-20,-2021

noahwsmith commented:
Corresponding VPAT: https://docs.google.com/document/d/1a1_DP4O3jjzB452UDAVrjOnGh1KzEJRR/edit

noahwsmith committed:
incremental improvements to install profile, including cleaning up sudo and initial homepage

noahwsmith committed:
clean up output from Make commands

noahwsmith committed:
freeze Context module commit hash

alxp commented:
I noticed that this PR is using islandora-sandbox along with the install profile repository. This means I was able to tell Ansible to use the project and profile. It runs up until the step to install openseadragon.

Then I noticed that the composer.json file in the install-profile branch of islandora-sandbox declares the requirement on the dev-main branch: https://github.com/Islandora-Devops/islandora-sandbox/blob/install-profile/composer.json#L69

```
"require": {
        "islandora/islandora_install_profile_demo": "dev-main"
    },
```

I see that the docker script uses git to clone the folder which is why the mismatch isn't caught.

I think the Sandbox project would make a good base for the aim we have been discussing for a common project.

noahwsmith committed:
changing the node owner for d9

noahwsmith committed:
adding composer_update

dannylamb committed:
make local works with the install profile

dannylamb committed:
Reverting make demo and local and renaming to -install-profile variants

dannylamb committed:
Updating TAG in sample.env

DonRichards commented:
@dannylamb Is this ready to test?

dannylamb commented:
Yes, `make demo-install-profile` and `make local-install-profile` should all bring you up a newly improved and d9 site with `islandora_install_profile_demo`.

Just make sure you set `ENVIRONMENT=local` or `ENVIRONMENT=demo` as appropriate in your .env file.

dannylamb commented:
fixed the merge conflict :sweat_smile:

DonRichards commented:
@dannylamb I was left more confused about where this is going. Are we trying to get this merged in or was there something else that we're doing?

dannylamb commented:
This depends on a lot of stuff.  If you merge it now it works, and `make local` and `make demo` are unchanged.  But there's `install_profile` and `bd-d9-updates` branches for various repos you'll want to merge in as well.

- https://github.com/Islandora-Devops/isle-buildkit/tree/install-profile
- https://github.com/Islandora-Devops/islandora_install_profile_demo/tree/bd-d9-update
- https://github.com/Islandora-Devops/islandora-sandbox/tree/install-profile
- https://github.com/Islandora-Devops/islandora_base_theme/tree/bd-d9-updates
- https://github.com/Islandora-Devops/islandora_starter_theme/tree/bd-d9-update

I'm swamped now but can make those PRs for those branches, though presumable everybody here as permissions to do that if they'd like. I didn't get around to making them b/c I wanted to QA it all a bit first since folks usually immediately descend onto a PR that isn't quite ready yet :sweat_smile:

dannylamb commented:
Also, just to be clear, this doesn't change `make local` or `make demo`.  All the new stuff is under `make demo-install-profile` and `make local-install-profile`.

DonRichards commented:
@dannylamb are you seeing during during the build this error? I ran `make demo-install-profile`

```log
In SolrCommandHelper.php line 41:               
  Unknown server               

search-api-solr:get-server-config [-h|--help] [-q|--quiet] [-v|vv|vvv|--verbose] [-V|--version] [--ansi] [--no-ansi] [-n|--no-interaction] [-d|--debug] [-y|--yes] [--no] [--remote-host REMOTE-HOST] [--remote-user REMOTE-USER] [-r|--root ROOT] [-l|--uri URI] [--simulate] [--pipe] [-D|--define DEFINE] [--xh-link XH-LINK] [--notify [NOTIFY]] [--druplicon] [--] <command> <server_id> [<file_name>] [<solr_version>]

unzip: can't open /tmp/ISLANDORA/solr_config.zip[.zip]
Waiting for 8983 on solr to open.
```

DonRichards commented:
@dannylamb To test this, am I expected to run `make demo` then `make demo-install-profile`?

rosiel commented:
> But there's install_profile and bd-d9-updates branches for various repos you'll want to merge in as well.
    https://github.com/Islandora-Devops/isle-buildkit/tree/install-profile
    https://github.com/Islandora-Devops/islandora_install_profile_demo/tree/bd-d9-update
    https://github.com/Islandora-Devops/islandora-sandbox/tree/install-profile
    https://github.com/Islandora-Devops/islandora_base_theme/tree/bd-d9-updates
    https://github.com/Islandora-Devops/islandora_starter_theme/tree/bd-d9-update

See:

* https://github.com/Islandora-Devops/isle-buildkit/pull/170
* https://github.com/Islandora-Devops/islandora_install_profile_demo/pull/5
* https://github.com/Islandora-Devops/islandora-sandbox/pull/5
* https://github.com/Islandora-Devops/islandora_base_theme/pull/2
* https://github.com/Islandora-Devops/islandora_starter_theme/pull/4

rosiel committed:
Git clone main branch of sandbox for local-install-profile.

rosiel committed:
Update buildkit commit in TAG variable.

Natkeeran committed:
Update homepage.json

Update homepage info

Natkeeran committed:
Update homepage.json

Natkeeran committed:
Update Makefile

adding  -T  option to run it via cron

DonRichards commented:
From scratch without any modifications running `make demo` results in many errors including `Failed to chmod file "/var/www/drupal/web/sites/default/settings.php". ` 

```terminal
  Command migrate:import was not found. Drush was unable to query the databas  
  e. As a result, many commands are unavailable. Re-run your command with --d  
  ebug to see relevant log messages.                                           
                                                                               

make[1]: *** [Makefile:150: run-islandora-migrations] Error 1
make[1]: Leaving directory '/home/admin/isle-dc'
make: *** [Makefile:322: demo] Error 2
```

noahwsmith commented:
@Natkeeran and @alxp does this match your understanding of where you left things in the last Sprint? What is the path forward here?

DonRichards commented:
Prior to testing I wiped out all things docker to test on a vanilla install. Testing was done on Linux machine running on the most recent commit #[003538d](https://github.com/Islandora-Devops/isle-dc/pull/160/commits/003538deef42101f45c4d60a40ae7816337bfc60)  
- Docker version 20.10.12, build e91ed57
- docker-compose version 1.29.2, build 5becea4c
Afterwords I also tested updating the sample.env locally prior to building with no difference. As a last ditch effort I tried to run it with codeserver knowing it builds the stack a little differently but again no luck.

DonRichards commented:
Also just tried with `Docker Compose version v2.2.3` in case the docker-composer version mattered but it didn't make any difference.

rosiel commented:
This code is intended to make `make local-install-profile` and `make demo-install-profile` work. In order to "not clobber" they were given separate commands, but originally these just overwrote `make local` and `make demo` with the install profile versions. But because of how `make demo` works, i don't think it's in a good state. It pulls the latest container from dockerhub that was populated by buildkit, and I think this is now the one designed for the install profile (since buildkit only builds one demo box and the newer ones were made for the install profile). 

I think that `make local` should work. In my personal opinion it should be fine to remove the original `make demo` and just let the install profile version be our demo.  it's been very lonely in th[e discussion of what the build options should be](https://github.com/Islandora/documentation/discussions/2037), so please feel free to chime in.

noahwsmith commented:
I support Rosie's proposal
> In my personal opinion it should be fine to remove the original make demo and just let the install profile version be our demo.

misilot commented:
To test this, do I need to use `TAG=aed5adc95784c1dd24589a0f77c3bbb063c92113` or can i use the latest isle-dc tagged version? I am trying to rebase it locally, so it is using the changes from the `development` branch.

rosiel commented:
Regarding the commit tag, Buildkit still has an 'install-profile' branch (and a [PR to merge it to main](https://github.com/Islandora-Devops/isle-buildkit/pull/170)) so I think it should be the most recent commit from that branch. 

Sorry bout the conflicts - seems like it should be okay to include all the new variables? Do they affect the install profile?

misilot commented:
I will take a look this afternoon (Central). Is there any chance that the Buildkit branch `install-profile` could be rebased to `main`? If not I can try using this branch as is without rebasing all the changes, and try it as is.

rosiel commented:
Rebase attempted. https://github.com/Islandora-Devops/isle-buildkit/pull/170

misilot commented:
Thanks @rosiel! 

I was able to successfully run `make local-install-profile` against `TAG=aed5adc95784c1dd24589a0f77c3bbb063c92113` with a rebased `install-profile` branch from `isle-dc`

I got the nice new green theme! :)

misilot commented:
I just ran the `make demo-install-profile` task and everything seemed to build happily.  I did notice that the homepage shows a 404.

<img width="1323" alt="image" src="https://user-images.githubusercontent.com/1446856/153475453-ec758973-d0ac-435b-af8a-7721b206491a.png">

misilot commented:
I did notice that when setting a custom domain the step for setting demo-data/homepage.json fails out

```
 [notice] Rolled back 13 items - done with 'islandora_defaults_tags'
make[1]: Entering directory '/klib/app/docker-compose/isle-dc'
curl -u admin:<PASSWORD> -H "Content-Type: application/json" -d "@demo-data/homepage.json" https://islandora.SERVER.net/node?_format=json
curl: (60) SSL: no alternative certificate subject name matches target host name 'islandora.SERVER.net'
More details here: https://curl.haxx.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.
make[1]: *** [Makefile:391: initial_content] Error 60
make[1]: Leaving directory '/klib/app/docker-compose/isle-dc'
make: *** [Makefile:384: local-install-profile] Error 2
```

I don't know if it would be better to have curl ignore the ssl cert errors during install?

noahwsmith commented:
Maybe that just needs a `-k` ? https://curl.se/docs/manpage.html#-k

misilot commented:
I noticed while running through `make local-install-profile` again that there was a couple of errors (listed below). Are these anything to be concerned about?

```
WARNING: Some services (activemq, alpaca, blazegraph, cantaloupe, crayfits, drupal, fcrepo, fits, homarus, houdini, hypercube, mariadb, matomo, milliner, recast, solr, traefik) use the 'deploy' key, which will be ignored. Compose does not support 'deploy' configuration - use `docker stack deploy` to deploy to a swarm.
--driver mysql
--host mariadb
--port 3306
--dbuser drupal_default
--dbpassword <password>
--dbname drupal_default
PROFILE: islandora_install_profile_demo
--account-mail=webmaster@localhost.com
--account-name=admin
--account-pass=1vagVTs86Cp6LkjjYVynHRwcZ8gdTQwzFXzlMQ1qiR1P1BDO
--site-mail=webmaster@localhost.com
--locale=en
--site-name=Default
--sites-subdir=default
USE_EXISTIG_CONFIG:
EVERYTHING ELSE:
Installing site.
-n si islandora_install_profile_demo --account-mail=webmaster@localhost.com --account-name=admin --account-pass=<password> --site-mail=webmaster@localhost.com --locale=en --site-name=Default --sites-subdir=default --db-url=mysql://drupal_default:<password>@mariadb:3306/drupal_default

 You are about to:
 * Create a sites/default/settings.php file
 * DROP all tables in your 'drupal_default' database.

 [notice] Starting Drupal installation. This takes a while.
 [notice] Performed install task: install_select_language
 [notice] Performed install task: install_select_profile
 [notice] Performed install task: install_load_profile
 [error]  Error importing/updating configuration
 [error]  Error importing/updating configuration
 [notice] Performed install task: install_verify_requirements
 [notice] Performed install task: install_settings_form
 [notice] Performed install task: install_verify_database_ready
 [notice] Performed install task: install_base_system
 [notice] Performed install task: install_bootstrap_full
 [warning] Could not find required jsonld.settings to add default RDF namespaces.
 [warning] The "views_block:members-block_1" was not found
 [error]  Error getting JWT token for message. Check JWT Configuration.
 [error]  Error getting JWT token for message. Check JWT Configuration.
 [notice] Performed install task: install_profile_modules
 [notice] Performed install task: install_profile_themes
 [notice] Performed install task: install_install_profile
 [notice] Performed install task: install_configure_form
 [notice] Performed install task: islandora_install_profile_demo_grant_shortcut_access
 [notice] Performed install task: islandora_install_profile_demo_set_default_theme
 [notice] Performed install task: islandora_install_profile_demo_finish_install
 [notice] Performed install task: install_finished
 [success] Installation complete.
```

noahwsmith commented:
My experience (backed up by @dannylamb) is that these errors are benign. It would be nice to eliminate them, but maybe we should acknowledge them in the docs until that's done?

DonRichards commented:
Something is up, I've tested this from a completely clean environment (cleared all docker cached images and configurations). Running `make local`, `make demo`, and `make local-install-profile` all resulting in several errors. All of which block it from fully building. I can run the `make local` and `make demo` from the __development__ branch without any issues.

misilot commented:
@DonRichards what I I ended up doing was rebasing this branch against development. Not sure if someone could do that in this PR and make it a little easier to test?

DonRichards commented:
After a rebase locally I'm seeing this during the build and I'm not completely sure what to look for sinse Defaults is included as well as sandbox. It does appear to me that this might be working.
This is a snippet of an error/warning I'm seeing.
```shell
...
Installing site.
-n si islandora_install_profile_demo --account-mail=webmaster@localhost.com --account-name=admin --account-pass=password --site-mail=webmaster@localhost.com --locale=en --site-name=Default --sites-subdir=default --db-url=mysql://drupal_default:password@mariadb:3306/drupal_default

 You are about to:
 * Create a sites/default/settings.php file
 * DROP all tables in your 'drupal_default' database.

 [notice] Starting Drupal installation. This takes a while.
 [notice] Performed install task: install_select_language
 [notice] Performed install task: install_select_profile
 [notice] Performed install task: install_load_profile
 [error]  Error importing/updating configuration 
 [error]  Error importing/updating configuration 
 [notice] Performed install task: install_verify_requirements
 [notice] Performed install task: install_settings_form
 [notice] Performed install task: install_verify_database_ready
 [notice] Performed install task: install_base_system
 [notice] Performed install task: install_bootstrap_full
[warning] Could not find required jsonld.settings to add default RDF namespaces.
 [warning] The "views_block:members-block_1" was not found
 [error]  Error getting JWT token for message. Check JWT Configuration. 
 [error]  Error getting JWT token for message. Check JWT Configuration.
```

## Should I push the rebase even with these warnings/errors?

rosiel commented:
If one of you is capable of pushing your rebase to this branch that would be great, I see it as something outside my skill level.

DonRichards commented:
@rosiel I'm just waiting on some confirmation just in case this messes someone else up.

DonRichards commented:
Well I guess 2 weeks is long enough. I'll start the rebase process.

noahwsmith commented:
@wgilling will be testing this today or early next week

wgilling commented:
Ok - here are my results today.  Success. I cloned the isle-dc project locally and changed to this `install-profile` branch, ran the `make demo-install-profile` and it completed without any failures and the resultant local website looks fine.

I added some taxonomy values for Family entities to refer to with one of the fields that is indexed by the default Solr config and was able to search for the related item based on the family name value. I even made a collection item and related the items to the collection and search and all other functionality worked as expected.

The only errors during the internal Makefile's `islandora_install_profile_demo` method were:
```
    [notice] Performed install task: install_load_profile
    [error] Error importing/updating configuration
    [error] Error getting JWT token for message. Check JWT Configuration.

```

I am not sure that those errors are critical or would needed addressed. Please let me know if any clarification is needed.

dannylamb commented:
If the site functions, then that can be ignored.  That function runs twice iirc, so one failure out of the two i think is normal.  TL;DR if you can make a derivative or put something in Fedora, then JWTs are properly configured and there's no issue.

noahwsmith committed:
Removing duplicate "confirm" function

noahwsmith commented:
Our testing found a duplicate function in the Makefile which I have now removed.

noahwsmith commented:
We've tested this again. Don's rebase is good, and we (Born-Digital) recommend that this be merged. I assume that someone else needs to hit the button, though?

rosiel commented:
I don't think I have any code in this, so I'll hit the button. Now if we can put some attention to https://github.com/Islandora-Devops/isle-buildkit/pull/170 ... and we'll be done.

noahwsmith commented:
YAY thanks Rosie!

misilot commented:
Thanks everyone!
